File: core/internet/intelligence/trust_engine.py
from __future__ import annotations

from core.internet.trust_engine import compute_trust


def score_trust(url: str, corroboration_count: int = 0, html_text: str = ""):
    """Explainable trust score — delegates to canonical compute_trust."""
    result = compute_trust(url, corroboration_count=corroboration_count)
    if html_text:
        from core.internet.citation_lineage_engine import reconstruct_citation_lineage

        cites = reconstruct_citation_lineage(html_text)
        result["citation_lineage"] = cites
        result["evidence"] = sorted(set(result.get("evidence", [])) + cites.get("evidence", []))
        result["deterministic_inputs"] = sorted(
            set(result.get("deterministic_inputs", [])) + [f"citations={cites['citation_count']}"]
        )
    return result

File: src/internet/intelligence/trustEngine.ts
/**
 * Converted from Python: core/internet/intelligence/trust_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../../runtime/pyCompat.js";
import { computeTrust } from "../trustEngine.js";
import { reconstructCitationLineage } from "../citationLineageEngine.js";

export function scoreTrust(url: any, corroboration_count: any = 0, html_text: any = ""): any {
  var result: any = computeTrust(url, corroboration_count);
  if (py.truthy(html_text)) {
    var cites: any = reconstructCitationLineage(html_text);
    py.setItem(result, "citation_lineage", cites);
    py.setItem(result, "evidence", py.sorted(py.add(py.toSet(py.get(result, "evidence", [])), py.get(cites, "evidence", []))));
    py.setItem(result, "deterministic_inputs", py.sorted(py.add(py.toSet(py.get(result, "deterministic_inputs", [])), [`citations=${py.toStr(py.at(cites, "citation_count"))}`])));
  }
  return result;
}
export { computeTrust, reconstructCitationLineage };
